lang/cs/Org.Apache.REEF.IMRU.Tests/TestActiveContextManager.cs

```diff
@@ -0,0 +1,287 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using Org.Apache.REEF.Driver.Context;
+using Org.Apache.REEF.Driver.Evaluator;
+using Org.Apache.REEF.IMRU.OnREEF.Driver;
+using Xunit;
+
+namespace Org.Apache.REEF.IMRU.Tests
+{
+    /// <summary>
+    /// Test methods in ContextManager
+    /// </summary>
+    public class TestActiveContextManager
```

Comment:
So I would like to point out that tests are not totally exhaustive. The exceptions in Add, Remove are not tested and in RemovedFailedContextInFailedEvaluator() although er test that exceptions are thrown we do not check the types as all cases throw same exception. On the other hand class is simple enough :) . So I will just make the observation here and leave it to you.